src/ktopt_planner_manager.cpp

```diff
@@ -29,26 +29,29 @@ class KTOptPlannerManager : public planning_interface::PlannerManager
   {
     robot_model_ = model;
     node_ = node;
-    parameter_namespace_ = parameter_namespace;
     param_listener_ = std::make_shared<ktopt_interface::ParamListener>(node, parameter_namespace);
-    //robot_description_subscriber_ = node_->create_subscription<std_msgs::msg::String>(
-    //    "robot_description", 10, std::bind(&KTOptPlannerManager::robot_description_callback, this, _1));
+
+
+  // set QoS to transient local to get messages that have already been published
+  // (if robot state publisher starts before planner)
     robot_description_subscriber_ = node_->create_subscription<std_msgs::msg::String>(
-      "/robot_description", rclcpp::SystemDefaultsQoS(), [this](const std_msgs::msg::String::SharedPtr msg){
-        if (!description_set_)
+      "robot_description", rclcpp::QoS(1).transient_local(), [this](const std_msgs::msg::String::SharedPtr msg){
```

Comment:
FYI: Setting `rclcpp::QoS(1).transient_local()` was the key because it allows the subsciber to read previously published messages and the robot description is only published when it is created/ changed